lib/shopify_api/session.rb

```diff
@@ -65,8 +66,9 @@ def encoded_params_for_signature(params)
       end
     end
 
-    def initialize(url, token = nil, extra = {})
+    def initialize(url, token = nil, api_version = :no_version, extra = {})
```

Comment:
Could `api_version` be a kwarg? Same q for `temp`.

Reply:
compatibility is my biggest concern with that.

Reply:
But couldn't a kwarg just have a default, too?

Reply:
could you provide the full signature you are picturing?

Reply:
(See Slack)

Reply:
Converged on "maybe it is better to just go to keyword args", i.e. `def initialize(url:, token:, api_version:, ...)`

Reply:
The method signature will change in the next PR, in which `api_version` becomes required.